osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs

```diff
@@ -25,31 +25,33 @@ protected DifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         /// <summary>
         /// Calculates the difficulty of the beatmap using a specific mod combination.
         /// </summary>
+        /// <param name="upTo">The time up to which hit objects should be considered.</param>
         /// <param name="mods">The mods that should be applied to the beatmap.</param>
         /// <returns>A structure describing the difficulty of the beatmap.</returns>
-        public DifficultyAttributes Calculate(params Mod[] mods)
+        public DifficultyAttributes Calculate(double upTo, params Mod[] mods)
```

Comment:
I'd create an overload for this, since you can't assign a default value to upTo.